app/gui2/src/components/GraphEditor.vue

```diff
@@ -213,6 +234,26 @@ const codeEditorHandler = codeEditorBindings.handler({
   },
 })
 
+function enterNode(id: ExprId) {
+  const expressionInfo = graphStore.db.getExpressionInfo(id)
+  if (expressionInfo == undefined || expressionInfo.methodCall == undefined) {
+    console.debug('Cannot enter node that has no method call.')
+    return
+  }
+  const definedOnType = tryQualifiedName(expressionInfo.methodCall.methodPointer.definedOnType)
+  if (definedOnType.ok && qnLastSegment(definedOnType.value) != 'Main') {
+    console.debug('Cannot enter node that is not defined on Main.')
+    return
+  }
```

Comment:
Instead of checking for `Main` as a last segment, this should be a check if the function is defined in any project-local module. For now, it could be a check if that matches the module that is currently opened (i.e. one from the top of the execution stack). Eventually we would support switching between project files, so that check could then be relaxed.